### PR TITLE
enable_cache_go_rule

### PR DESCRIPTION
**Pull request by SanaaYousaf**

### CODE REVIEW

**Comment by liamg** [comment]
Hi @SanaaYousaf - thanks for the PR. Do you think there is a demonstrable security impact of leaving the cache disabled? If so could you add a bit more info about this to the rule metadata under "explanation/impact"?

File: internal/rules/aws/apigateway/enable_cache.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+)
+
+var CheckEnableCache = rules.Register(
+	scan.Rule{
+		AVDID:      "AVD-AWS-0179",
+		Provider:   providers.AWSProvider,
+		Service:    "api-gateway",
+		ShortCode:  "enable-cache",
+		Summary:    "Ensure that response caching is enabled for your Amazon API Gateway REST APIs.",
+		Impact:     "",
```
Comment (AkhtarAmir) [inline internal/rules/aws/apigateway/enable_cache.go:18]:
```suggestion
		Impact:     "Reduce the number of calls made to your API endpoint and also improve the latency of requests to your API with response caching.",
```

File: internal/rules/aws/apigateway/enable_cache.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	scan.Rule{
+		AVDID:      "AVD-AWS-0190",
+		Provider:   providers.AWSProvider,
+		Service:    "api-gateway",
+		ShortCode:  "enable-cache",
+		Summary:    "Ensure that response caching is enabled for your Amazon API Gateway REST APIs.",
+		Impact:     "Reduce the number of calls made to your API endpoint and also improve the latency of requests to your API with response caching.",
+		Resolution: "Enable cache",
+		Explanation: "A REST API in API Gateway is a collection of resources and methods that are integrated with backend HTTP endpoints, Lambda functions, or other AWS services.You can enable API caching in Amazon API Gateway to cache your endpoint responses." +
+			"With caching, you can reduce the number of calls made to your endpoint and also improve the latency of requests to your API.",
```
Comment (simar7) [inline internal/rules/aws/apigateway/enable_cache.go:21]:
```suggestion
		Explanation: "A REST API in API Gateway is a collection of resources and methods that are integrated with backend HTTP endpoints, Lambda functions or other AWS services. You can enable API caching in Amazon API Gateway to cache your endpoint responses. With caching, you can reduce the number of calls made to your endpoint and also improve the latency of requests to your API.",
```

File: internal/rules/aws/apigateway/enable_cache.tf.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+ resource "aws_api_gateway_method_settings" "bad_example" {
+   rest_api_id = aws_api_gateway_rest_api.example.id
+   stage_name  = aws_api_gateway_stage.example.stage_name
+   method_path = "path1/GET"
+ 
+   settings {
+     metrics_enabled = true
+     logging_level   = "INFO"
+     caching_enabled = true
```
Comment (simar7) [inline internal/rules/aws/apigateway/enable_cache.tf.go:45]: Shouldn't this be missing here (to signify the cache being disabled)?

File: internal/rules/aws/apigateway/enable_cache_test.go [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+	defsecTypes "github.com/aquasecurity/defsec/pkg/types"
+
+	v1 "github.com/aquasecurity/defsec/pkg/providers/aws/apigateway/v1"
+
+	"github.com/aquasecurity/defsec/pkg/state"
+
+	"github.com/aquasecurity/defsec/pkg/scan"
+
+	"github.com/stretchr/testify/assert"
```
Comment (simar7) [inline internal/rules/aws/apigateway/enable_cache_test.go:14]: could you make sure the imports are sorted? `go import` should take care of it or your IDE of choice.

**Review by simar7 — CHANGES_REQUESTED** [review]
looks good, left a couple of comments.

File: internal/rules/aws/apigateway/enable_cache_test.go (hunk 4 above)
Comment (SanaaYousaf) [reply internal/rules/aws/apigateway/enable_cache_test.go:14]: current rules are not following this practice i.e. imports are not sorted. So should we keep it consistent?

**Comment by simar7** [comment]
Can you also generate the avd docs for this? Run `make docs` and commit the generated docs with this PR.

**Review by simar7 — APPROVED** [review]
thanks lgtm

**Pull request merged**